### ISSUE TITLE
Autoclose brackets etc

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#802]
In several editors I have come to expect that when I enter one of these character `( [ {` the matching closing item is added automatically. When I *then* hit `Enter` the block is automatically formatted: add an indented empty line, put the closing item on the next line and put the cursor in the indented line.

Some editors even do that properly when there is a selection and I enter an opening bracket. That is, they don't overwrite the selection but properly format it in a block of brackets.

I think this would be nice for Frescobaldi too, at least for the curly braces. Nearly always when one writes `{` one will want to  produce a new block.

What do you think?

**Comment by PeterBjuhr** [comment]
I agree that Frescobaldi could be more helpful with these things. But when adding things automatically I think there should be a way to turn it off for those how find it annoying.

**Comment by uliska** [comment]
Yes, definitely such functionality should be governed by Preferences

**Comment by wbsoft** [comment]
This would certainly be helpful.
